optimum/intel/openvino/trainer.py

```diff
@@ -719,8 +719,10 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
                     if not save_as_external_data:
                         onnx_path.write_bytes(f.getvalue())
                     logger.warning(
-                        f"Error encountered during IR pruning: {err}. {onnx_path} is dumped for debug. Run continues."
+                        f"Error encountered during OpenVINO IR pruning: {err}. {onnx_path} is dumped for debugging. "
```

Comment:
I expect most users will not know what to do with the ONNX model for debugging. It's a blanket exception, it's a bit risky to only show a warning. Shouldn't we reraise the exception and then still dump the ONNX model? Then it's at least clear that something is broken, and you get the exact error, instead of just a generic warning that may be easy to miss.